src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseImportBasicService.java

```diff
@@ -285,6 +284,43 @@ private void setupExerciseForImport(ProgrammingExercise newExercise) {
         }
     }
 
+    /**
+     * Disconnect child entities from the exercise.
+     * <p>
+     * Just setting the collections to {@code null} breaks the automatic orphan removal in the database.
+     *
+     * @param newExercise The new exercise that should be created during import.
+     */
+    private void disconnectRelatedEntities(final ProgrammingExercise newExercise) {
+        if (newExercise.getExerciseHints() != null) {
+            newExercise.getExerciseHints().clear();
+        }
+        if (newExercise.getTestCases() != null) {
+            newExercise.getTestCases().clear();
+        }
+        if (newExercise.getStaticCodeAnalysisCategories() != null) {
+            newExercise.getStaticCodeAnalysisCategories().clear();
+        }
+        if (newExercise.getAttachments() != null) {
+            newExercise.getAttachments().clear();
+        }
+        if (newExercise.getPlagiarismCases() != null) {
+            newExercise.getPlagiarismCases().clear();
+        }
+        if (newExercise.getTutorParticipations() != null) {
+            newExercise.getTutorParticipations().clear();
+        }
+        if (newExercise.getExampleSubmissions() != null) {
+            newExercise.getExampleSubmissions().clear();
+        }
+        if (newExercise.getPosts() != null) {
+            newExercise.getPosts().clear();
+        }
```

Comment:
This all should be part of the exercise class (from my point of view)

Reply:
Good point, done in 8393aaaab3191224a5e7fb190df924d48a779f6d.

That whole disconnect logic was introduced to fix an issue exactly like the SQL one you posted below. I now extended the disconnect to properly clear all related connections, some of which were missed before. Maybe that helps, I’ve not tested that yet.